browser/ui/views/location_bar/brave_news_location_view.cc

```diff
@@ -10,210 +10,154 @@
-                                          base::Unretained(this)));
-    opted_in_.Init(brave_news::prefs::kBraveTodayOptedIn, profile->GetPrefs(),
-                   base::BindRepeating(&BraveNewsButtonView::Update,
-                                       base::Unretained(this)));
-    news_enabled_.Init(brave_news::prefs::kNewTabPageShowToday,
-                       profile->GetPrefs(),
-                       base::BindRepeating(&BraveNewsButtonView::Update,
-                                           base::Unretained(this)));
-
-    auto menu_button_controller = std::make_unique<views::MenuButtonController>(
-        this,
-        base::BindRepeating(&BraveNewsButtonView::ButtonPressed,
-                            base::Unretained(this)),
-        std::make_unique<views::Button::DefaultButtonControllerDelegate>(this));
-    SetButtonController(std::move(menu_button_controller));
-
-    Update();
+BraveNewsLocationView::BraveNewsLocationView(
+    Profile* profile,
+    IconLabelBubbleView::Delegate* icon_label_bubble_delegate,
+    PageActionIconView::Delegate* page_action_icon_delegate)
+    : PageActionIconView(nullptr,
+                         0,
+                         icon_label_bubble_delegate,
+                         page_action_icon_delegate,
+                         "BraveNewsFollow") {
+  SetLabel(l10n_util::GetStringUTF16(IDS_BRAVE_NEWS_ACTION_VIEW_TOOLTIP));
+
+  last_contents_ = GetWebContents();
+  if (last_contents_) {
+    BraveNewsTabHelper::FromWebContents(last_contents_)->AddObserver(this);
   }
-  ~BraveNewsButtonView() override = default;
-  BraveNewsButtonView(const BraveNewsButtonView&) = delete;
-  BraveNewsButtonView& operator=(const BraveNewsButtonView&) = delete;
-
-  void Update() {
-    if (!should_show_.GetValue() || !news_enabled_.GetValue() ||
-        !opted_in_.GetValue()) {
-      SetVisible(false);
-      return;
-    }
 
-    auto* contents = tab_strip_->GetActiveWebContents();
-    bool subscribed = false;
-    bool has_feeds = false;
-    absl::optional<BraveNewsTabHelper::FeedDetails> feed;
+  should_show_.Init(brave_news::prefs::kShouldShowToolbarButton,
+                    profile->GetPrefs(),
+                    base::BindRepeating(&BraveNewsLocationView::UpdateImpl,
+                                        base::Unretained(this)));
+  opted_in_.Init(brave_news::prefs::kBraveTodayOptedIn, profile->GetPrefs(),
+                 base::BindRepeating(&BraveNewsLocationView::UpdateImpl,
+                                     base::Unretained(this)));
+  news_enabled_.Init(brave_news::prefs::kNewTabPageShowToday,
+                     profile->GetPrefs(),
+                     base::BindRepeating(&BraveNewsLocationView::UpdateImpl,
+                                         base::Unretained(this)));
+
+  Update();
+}
+
+BraveNewsLocationView::~BraveNewsLocationView() = default;
+
+views::BubbleDialogDelegate* BraveNewsLocationView::GetBubble() const {
```

Comment:
nit: GetBubbleView

Reply:
This is an overridden method of the base class. Its name has to stay the same.